### PR TITLE
Improve websocket error handling and fix #315

### LINKED ISSUE #315: Error: read ECONNRESET
Hello, I am trying to get started with Parcel. I created an index.html file with an index.js and when I run

`parcel index.html`

localhost:1234 opens fine but as soon as I save the file the server stops and I get

```
events.js:136
      throw er; // Unhandled 'error' event
      ^

Error: read ECONNRESET
    at _errnoException (util.js:999:13)
    at TCP.onread (net.js:629:25)
```

### PR DESCRIPTION
**Pull request by DeMoorJasper**
Logs errors accuring while sending data instead of throwing them as unhandled errors.
Also added an ignore to ECONNRESET, because this gets triggered every time html reload is triggered.
Closes #315

TODO: 
* log errors using Logger, related to: #207 
* ~~Fix hmr related tests~~ [DONE]

### CODE REVIEW

**Comment by FalkoJoseph** [comment]
@DeMoorJasper Does this fix the issue where Parcel crashes on refreshing the page? (using parcel-plugin-vue)

**Comment by DeMoorJasper** [comment]
@FalkoJoseph Yup it does

**Comment by FalkoJoseph** [comment]
@DeMoorJasper Nice, can't wait! Also cool to see a Howest student contributing to this project 👍

**Comment by DeMoorJasper** [comment]
@devongovett could we get this PR approved and released as a bugfix, lot's of people are being affected with clean install #315

File: src/HMRServer.js [hunk 1]
```diff
@@ -69,6 +72,15 @@ class HMRServer {
     }
   }
 
+  handleSocketError(err) {
+    if (err.code === 'ECONNRESET') {
+      // This gets triggered on page refresh, ignore this
+      return;
+    }
+    // TODO: Use logger to print errors
+    console.log(prettyError(err));
```
Comment (devongovett) [inline src/HMRServer.js]: have you seen any other errors actually happen? should we just ignore them?

**Comment by devongovett** [comment]
Published in v1.2.1.

File: src/HMRServer.js (hunk 1 above)
Comment (DeMoorJasper) [reply src/HMRServer.js]: This was just for future error handling when adding new features.
Normally ECONNRESET is the only error that accures due to chrome not properly closing the connection

File: src/HMRServer.js [hunk 2]
```diff
@@ -8,11 +8,14 @@ class HMRServer {
     });
 
     this.wss.on('connection', ws => {
+      ws.onerror = this.handleSocketError;
       if (this.unresolvedError) {
         ws.send(JSON.stringify(this.unresolvedError));
       }
     });
 
+    this.wss.on('error', this.handleSocketError);
```
Comment (lpinca) [inline src/HMRServer.js]: This is probably not needed as it adds the listener after the server is bound. errors are forwarded from the internal `http.Server` and most of them (all?) are emitted when listening.

File: src/HMRServer.js (hunk 1 above)
Comment (lpinca) [reply src/HMRServer.js]: Other errors might be parser errors, like an invalid frame, an inflate error if you use permessage-deflate, a frame with a payload bigger than the `maxPayload` option or other `net.Socket` errors like write EPIPE, write ECONNABORTED, etc.

**Pull request merged**